src/config/backends/raspberry-pi.ts

```diff
@@ -0,0 +1,157 @@
+import * as _ from 'lodash';
+import { fs } from 'mz';
+
+import {
+	ConfigOptions,
+	DeviceConfigBackend,
+	bootMountPoint,
+	remountAndWriteAtomic,
+} from './backend';
+import * as constants from '../../lib/constants';
+import log from '../../lib/supervisor-console';
+
+/**
+ * A backend to handle Raspberry Pi host configuration
+ *
+ * Supports:
+ * 	- {BALENA|RESIN}_HOST_CONFIG_dtparam = value | "value" | "value1","value2"
+ * 	- {BALENA|RESIN}_HOST_CONFIG_dtoverlay = value | "value" | "value1","value2"
+ * 	- {BALENA|RESIN}_HOST_CONFIG_device_tree_param = value | "value" | "value1","value2"
+ * 	- {BALENA|RESIN}_HOST_CONFIG_device_tree_overlay = value | "value" | "value1","value2"
+ * 	- {BALENA|RESIN}_HOST_CONFIG_gpio = value | "value" | "value1","value2"
+ */
+
+export class RPiConfigBackend extends DeviceConfigBackend {
+	private static bootConfigVarPrefix = `${constants.hostConfigVarPrefix}CONFIG_`;
+	private static bootConfigPath = `${bootMountPoint}/config.txt`;
+
+	public static bootConfigVarRegex = new RegExp(
+		'(' + _.escapeRegExp(RPiConfigBackend.bootConfigVarPrefix) + ')(.+)',
```

Comment:
Since you only seem to care about the second group you could make the first non-capturing like
```suggestion
		'(?:' + _.escapeRegExp(RPiConfigBackend.bootConfigVarPrefix) + ')(.+)',
```
and then use `$1` in the replacement instead

Reply:
added